File: Aula03/InAula/Carro.py

from Veiculo import Veiculo

class Carro(Veiculo):
    def __init__(self, marca, modelo, ano, placa, numeroPortas):
        super().__init__(marca, modelo, ano, placa)
        self.numeroPortas = numeroPortas
    
    def __str__(self):
        return super().__str__() + f"\nNúmero de Portas: {self.numeroPortas}"

File: Aula11_JavaHeranca/Carro.java
public class Carro extends Veiculo {
    private int numeroPortas;
    
    @Override
    public double calcularImposto() {
        return 1000.0;
    }


    @Override
    public String toString() {
        return super.toString()
            + " - N. Portas: " + this.getNumeroPortas();
    }

    public int getNumeroPortas() {
        return numeroPortas;
    }
    public void setNumeroPortas(int numeroPortas) {
        this.numeroPortas = numeroPortas;
    }
    public Carro(String marca, String modelo, int ano, String placa, int numeroPortas) {
        super(marca, modelo, ano, placa);
        this.numeroPortas = numeroPortas;
    }

   
    
}
